lib/ExecutionEngine/ExecutionEngine.cpp

```diff
@@ -130,7 +151,15 @@ void glow::runBatch(ExecutionEngine &EE, Context &ctx, size_t iterations,
 
 void ExecutionEngine::compile(CompilationMode mode, Function *F) {
   backend_->optimizeFunction(mode, F);
-  function_ = backend_->compile(F);
+  compiledFunctions_.clear();
```

Comment:
this behavior needs to be in the comments. `compile` essentially discards all previously compiled functions.